phpBB/phpbb/profilefields/type/type_bool.php

```diff
@@ -405,11 +405,14 @@ public function display_options(&$template_vars, &$field_data)
 			$field_data['lang_options'][1] = '';
 		}
 
-		$template_vars = array_merge($template_vars, array(
-			'S_BOOL'					=> true,
-			'L_LANG_OPTIONS_EXPLAIN'	=> $this->user->lang['BOOL_ENTRIES_EXPLAIN'],
-			'FIRST_LANG_OPTION'			=> $field_data['lang_options'][0],
-			'SECOND_LANG_OPTION'		=> $field_data['lang_options'][1],
-		));
+		$doptions = array(
+			0 => array(
+					'TITLE' => $this->user->lang['ENTRIES'],
+					'EXPLAIN' => $this->user->lang['BOOL_ENTRIES_EXPLAIN'],
+					'FIELD' => '<input class="medium" id="lang_options" name="lang_options[0]" value="' . $field_data['lang_options'][0] . '" /> ' . $this->user->lang['FIRST_OPTION'] . '</dd><dd><input class="medium" name="lang_options[1]" value="' . $field_data['lang_options'][1] . '" /> ' . $this->user->lang['SECOND_OPTION']
```

Comment:
I am doing the same as here:
https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/profilefields/type/type_bool.php#L85

I agree that that should be the goal, but I think that would need a bigger change in the profile fields code.  Also affecting the language specific part (step 3).

A similar schema as used here might be possible as well:
https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/profilefields/type/type_base.php#L185-L205